src/Catalog/CatalogWriterBase.cs

```diff
@@ -183,7 +191,7 @@ protected virtual StorageContent CreateIndexContent(IGraph graph, Uri type)
         {
             IGraph graph = new Graph();
 
-            INode resourceNode = graph.CreateUriNode(resourceUri);
+            INode resourceNode = graph.CreateUriNode(CatalogItem.AliasesAddress(resourceUri, _customDomain));
```

Comment:
We discussed offline and I agree that the changes could be risky.  I tried to be as less intrusive as possible  using the current design. The mental model that I followed was the separation of concerns between the storage and content. The host links in content (generated by the writers) can be different by the storage host. We can do more radical design changes but  that it will add to the time and risk.